### PR TITLE
Add the section to the checklist: List of requirements for collections to be included in Ansible

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
Related to https://github.com/ansible/community/issues/539#issuecomment-729954216

Add the section to the checklist: List of requirements for collections to be included in Ansible

I know that it hasn't been discussed yet. However **we should start from something** and, in my opinion, there are obvious things listed in the PR. Anyway, i can correct / remove something if it's too premature now.

Alternatively, we could start listing the requirements in a separate file not mixing everything in one place.

Note: i put links to the doc pages which is in devel because they are actually latest, can be important in this case.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
`collection_requirements.rst`

### CODE REVIEW

**Comment by Andersson007** [comment]
cc @gundalow @felixfontein @abadger @dmsimard @dericcrago

File: collection_requirements.rst [hunk 1]
```diff
@@ -191,6 +191,19 @@ Also:
...
+==============================================================
+
+To be included in the `ansible` package, collections must meet the following criteria:
+
+* `development conventions <https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html>`_
+* `Collection requirements <https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst>`_ (this document)
+* `Ansible documentation format <https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html>`_ and the `style guide <https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide>`_
+* to pass the Ansible `sanity tests <https://docs.ansible.com/ansible/devel/dev_guide/testing_sanity.html#testing-sanity>`_
+* to have `unit <https://docs.ansible.com/ansible/devel/dev_guide/testing_units.html#testing-units>`_ and / or `integration tests <https://docs.ansible.com/ansible/devel/dev_guide/testing_integration.html#testing-integration>`_
+* Public bug tracker and accept Pull Requests.
```
Comment (felixfontein) [inline collection_requirements.rst:204]: BTW, right now we include at least one collection which does not accept pull requests.

Comment (felixfontein) [reply collection_requirements.rst:204]: Also, we have a discussion ticket for some of these points: https://github.com/ansible-collections/overview/issues/126

Comment (Andersson007) [reply collection_requirements.rst]: I removed this point for a while

**Comment by Andersson007** [comment]
ready_for_review

File: collection_requirements.rst [hunk 2]
```diff
@@ -182,7 +182,7 @@ All related entities must be moved / copied including:
 * CI and unit tests
 * corresponding documentation fragments from plugins/doc_fragments
 
-Also:
+Also:requi
```
Comment (gundalow) [inline collection_requirements.rst:185]:
```suggestion
Also:
```

File: collection_requirements.rst [hunk 3]
```diff
@@ -191,6 +191,18 @@ Also:
 See `Migrating content to a different collection <https://docs.ansible.com/ansible/devel/dev_guide/developing_collections.html#migrating-ansible-content-to-a-different-collection>`_ for complete details.
 
 
+List of requirements for collections to be included in Ansible
+==============================================================
```
Comment (gundalow) [inline collection_requirements.rst:195]:
```suggestion
Requirements for collections to be included in the Ansible Package
==================================================================
```

File: collection_requirements.rst [hunk 4]
```diff
@@ -191,6 +191,18 @@ Also:
...
+
+To be included in the `ansible` package, collections must meet the following criteria:
+
+* `development conventions <https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html>`_
+* `Collection requirements <https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst>`_ (this document)
+* `Ansible documentation format <https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html>`_ and the `style guide <https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide>`_
+* to pass the Ansible `sanity tests <https://docs.ansible.com/ansible/devel/dev_guide/testing_sanity.html#testing-sanity>`_
+* to have `unit <https://docs.ansible.com/ansible/devel/dev_guide/testing_units.html#testing-units>`_ and / or `integration tests <https://docs.ansible.com/ansible/devel/dev_guide/testing_integration.html#testing-integration>`_
```
Comment (dericcrago) [inline collection_requirements.rst:203]: when I click on the links, they take me directly to the content and just *below* the header, would it be better to use the header links for maybe a bit more context when landing on the page?

```suggestion
* to have `unit <https://docs.ansible.com/ansible/devel/dev_guide/testing_units.html#unit-tests>`_ and / or `integration tests <https://docs.ansible.com/ansible/devel/dev_guide/testing_integration.html#integration-tests>`_
```

**Review by dericcrago — COMMENTED** [review]
LGTM, thanks @Andersson007

File: collection_requirements.rst (hunk 4 above)
Comment (Andersson007) [reply collection_requirements.rst]: @dericcrago ok, sounds good, thanks!

Comment (Andersson007) [reply collection_requirements.rst]: committed

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
@Andersson007 Thank you for this
@felixfontein @dericcrago Thank you for the reviews

**Comment by Andersson007** [comment]
thanks everyone!

**Pull request merged**